src/carplay.cs

```diff
@@ -0,0 +1,720 @@
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPSessionConfiguration {
+
+		[Export ("initWithDelegate:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ICPSessionConfigurationDelegate @delegate);
+
+		[Export ("limitedUserInterfaces")]
+		CPLimitableUserInterface LimitedUserInterfaces { get; }
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ICPSessionConfigurationDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+	}
+
+	interface ICPSessionConfigurationDelegate { }
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model ( AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPSessionConfigurationDelegate {
+
+		[Abstract]
+		[Export ("sessionConfiguration:limitedUserInterfacesChanged:")]
+		void LimitedUserInterfacesChanged (CPSessionConfiguration sessionConfiguration, CPLimitableUserInterface limitedUserInterfaces);
+	}
+
+	[Abstract]
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	interface CPTemplate : NSSecureCoding {
+
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPRouteChoice : NSCopying, NSSecureCoding {
+
+		[Export ("initWithSummaryVariants:additionalInformationVariants:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string [] summaryVariants, string [] additionalInformationVariants);
+
+		[Export ("summaryVariants", ArgumentSemantic.Copy)]
+		string [] SummaryVariants { get; }
+
+		[NullAllowed, Export ("additionalInformationVariants", ArgumentSemantic.Copy)]
+		string [] AdditionalInformationVariants { get; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPTrip : NSSecureCoding {
+
+		[Export ("initWithOrigin:destination:routeChoices:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MKMapItem origin, MKMapItem destination, CPRouteChoice [] routeChoices);
+
+		[Export ("origin", ArgumentSemantic.Strong)]
+		MKMapItem Origin { get; }
+
+		[Export ("destination", ArgumentSemantic.Strong)]
+		MKMapItem Destination { get; }
+
+		[Export ("routeChoices", ArgumentSemantic.Copy)]
+		CPRouteChoice [] RouteChoices { get; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	[DisableDefaultCtor]
+	interface CPVoiceControlTemplate {
+
+		[Export ("initWithTitleVariants:animatedImage:")]
+		IntPtr Constructor (string [] titleVariants, UIImage animatedImage);
+
+		[NullAllowed, Export ("titleVariants", ArgumentSemantic.Copy)]
+		string [] TitleVariants { get; }
+
+		[NullAllowed, Export ("animatedImage", ArgumentSemantic.Strong)]
+		UIImage AnimatedImage { get; }
```

Comment:
Same: `NullAllowed` doesn't match the corresponding parameter in the constructor.

Reply:
yep, agreed, according to docs this should not be nil since it is a requirement in the ctor